src/mongo/mongoConnectionStrings.ts

```diff
@@ -31,6 +41,9 @@ export function getDatabaseNameFromConnectionString(connectionString: string): s
 
 export function addDatabaseToAccountConnectionString(connectionString: string, databaseName: string): string | undefined {
     try {
+        if (connectionString.includes(ext.emulatorPassword)) {
```

Comment:
You have logic in both `getDatabaseNameFromConnectionString` and `addDatabaseToAccountConnectionString ` to make sure the emulator connection string works. However, I think you should move this logic up the stack so that the connection string on the account tree item is always correct _before_ it gets passed to either of these functions (or anywhere else it's used).

Also I like the replace strategy better than what you did above, but I would suggest this:
```
connectionString.replace(emulatorPassword, encodeUriComponent(emulatorPassword))
```

Reply:
I'm trying to understand what correct would mean here. Andrii's code would not work because it looks to separate by slashes - that's fine for all cases but the emulator password (`C2y6yDjf5/R+...`). To that end, I don't see how I can sanitize the connection string to  work with the current code.

Reply:
Correct means you've encoded the password (using encodeUriComponent) so that there are no slashes